src/main/java/spoon/reflect/declaration/CtElement.java

```diff
@@ -391,4 +392,18 @@
 	 */
 	List<CtElement> getDirectChildren();
 
+	/**
+	 * @return the source code of this element, with fully-qualified types.
+	 *
+	 * {@link Environment#getToStringMode()} is not taken into account
+	 *
+	 * Use {@link #prettyprint()} for a nice source code.
+	 */
+	String toString();
+
+	/**
+	 * @return the source code of this element accorfing to {@link Environment#getToStringMode()}.
+	 */
```

Comment:
@monperrus , I suggest to add there note like: "Call of prettyprint() method can modify spoon model! It will apply model import validators, which will try to detect correct import statements, detect conflicts, change implicitness of types and packages, etc."